### DISCUSSION TITLE
Does Strimzi mirrormaker2 provide licensed support

### DISCUSSION THREAD

**Discussion by yogeshgoel001** [strimzi/strimzi-kafka-operator#10724] [General]
Does Strimzi mirrormaker2 has professional support available??

**Comment by scholzj** [comment]
Strimzi is an open-source project/community. Not a company. So it does not provide any commercial support. But it is used as part of products offered by companies such as Red Hat, IBM, or Axual (and possibly others).

**Reply by yogeshgoel001** [reply]
ohk, Does strimizi controller stores the connection endpoint in cache, as If I change the connection endpoint in Mirrormaker2 it restarts the pod, but still it keeps on pointing to the old one for sometime with WARN (Can not connect due to authentication issue). Can we remove these Warnings ??

**Reply by scholzj** [reply]
Not sure how that relates to the initial question. And not sure what Strimzi Controller is - you mean the operator? You should probably start by providing a full scenario, configuration and full logs.

**Reply by yogeshgoel001** [reply]
Scenario. 
We have Primary (Event Hub) and DR (Event Hub) and Primary (CFK) (as destination)
Initially Mirrormaker2 pointing was from Primary (event hub) --> Primary (CFK) and was mirroring the data properly. 
Then we changed the Mirrormaker2 pointing from DR (Event Hub) --> Primary (CFK)
After changing the pointing we faced 2 issues. 
1.) Mirrormaker2 pod was still showing Primary (Event HUB) details after self-restart as WARN (Authentication failed)
2.) Mirroring between DR(Event HUB) --> Primary (CFK) didn't work. 

Hope the scenario is clear. 
Can you please help here?

**Reply by scholzj** [reply]
No, you have to share the configurations, you have to share the logs, the exact steps you did and when. Without that nobody can tell what is happening.

**Reply by yogeshgoel001** [reply]
Current configurations:
  - alias: kafka
    authentication:
      certificateAndKey:
        certificate: tls.crt
        key: tls.key
        secretName: secret-tls
      type: tls
    bootstrapServers: kafka-service-qe-prod-backingservices-0.testgrp1.attbsse.amdocs.com:9580
    config:
      auto.commit.enabled: false
      auto.commit.interval.ms: 10000
      auto.offset.reset: latest
      config.storage.replication.factor: 1
      config.storage.topic: bsse-cfk-cluster-configs
      consumer.connections.max.idle.ms: 18000
      consumer.metadata.max.age.ms: 18000
      offset.storage.replication.factor: 1
      offset.storage.topic: bsse-cfk-cluster-offsets
      ssl.cipher.suites: TLS_ECDHE_RSA_WITH_AES_256_GCM_SHA384
      ssl.enabled.protocols: TLSv1.2
      ssl.protocol: TLSv1.2
      status.storage.replication.factor: 1
      status.storage.topic: bsse-cfk-cluster-status
    tls:
      trustedCertificates:
      - certificate: amdocs.crt
        secretName: secret-tls
  - alias: ehubwus2
    authentication:
      passwordSecret:
        password: eventhubspassword
        secretName: westus2eventhubssecret
      type: plain
      username: $ConnectionString
    In
    config:
      acks: -1
      block.on.buffer.full: true
      config.storage.replication.factor: 1
      max.in.flight.requests.per.connection: 1
      status.storage.replication.factor: 1
    tls:
      trustedCertificates: []
  connectCluster: kafka
  livenessProbe:
    initialDelaySeconds: 15
    timeoutSeconds: 5
  metricsConfig:
    type: jmxPrometheusExporter
    valueFrom:
      configMapKeyRef:
        key: metrics-config.yml
        name: mirror-maker-2-metrics


Error:
2024-10-16 15:24:37,990 WARN [ehub->kafka.MirrorCheckpointConnector|worker] [AdminClient clientId=ehub->kafka|ehub->kafka.MirrorCheckpointConnector|checkpoint-source-admin] Connection to node -1 (**ctx-ehub-eu2-dev-test-shared-ns-01.servicebus.windows.net**/10.164.28.20:9093) terminated during authentication. This may happen due to any of the following reasons: (1) Authentication failed due to invalid credentials with brokers older than 1.0.0, (2) Firewall blocking Kafka TLS traffic (eg it may only allow HTTPS traffic), (3) Transient network issue. (org.apache.kafka.clients.NetworkClient) [kafka-admin-client-thread | ehub->kafka|ehub->kafka.MirrorCheckpointConnector|checkpoint-source-admin]
2024-10-16 15:24:38,188 INFO 10.164.1.7 - - [16/Oct/2024:15:24:38 +0000] "GET /connectors HTTP/1.1" 200 116 "-" "-" 2 (org.apache.kafka.connect.runtime.rest.RestServer) [qtp281174369-32]
2024-10-16 15:24:38,304 WARN [ehub->kafka.MirrorSourceConnector|worker] [AdminClient clientId=ehub->kafka|ehub->kafka.MirrorSourceConnector|offset-syncs-source-admin] Unexpected error from ctx-ehub-eu2-dev-test-shared-ns-01.servicebus.windows.net/10.164.28.20 (channelId=-1); closing connection (org.apache.kafka.common.network.Selector) [kafka-admin-client-thread | ehub->kafka|ehub->kafka.MirrorSourceConnector|offset-syncs-source-admin]

Steps: 
1.) Firstly below configuration was, which was working fine. 
bootstrapServers: ctx-ehub-eu2-dev-test-shared-ns-01.servicebus.windows.net:9093
2.) Then changed this to 
bootstrapServers: ctx-ehub-wu2-dev-test-shared-ns-01.servicebus.windows.net:9093

after this change it started throwing the above error.

**Reply by yogeshgoel001** [reply]
can you please check this once??

**Reply by scholzj** [reply]
You have to share the full configurations, full logs from all componetnts and the steps that you did. Without that, nobody can tell you what really happened.
